internal/events/blockchain_event_test.go

```diff
@@ -152,31 +152,53 @@ func TestContractEventWrongNS(t *testing.T) {
 }
 
 func TestPersistBlockchainEventDuplicate(t *testing.T) {
-	em := newTestEventManager(t)
-	defer em.cleanup(t)
+	type testCase struct {
+		name            string
+		event           *core.BlockchainEvent
+		existingID      *fftypes.UUID
+		expectedID      *fftypes.UUID
+		expectedError   error
+		expectedCreated bool
+	}
 
-	ev := &core.BlockchainEvent{
-		ID:         fftypes.NewUUID(),
-		Name:       "Changed",
-		Namespace:  "ns1",
-		ProtocolID: "10/20/30",
-		Output: fftypes.JSONObject{
-			"value": "1",
-		},
-		Info: fftypes.JSONObject{
-			"blockNumber": "10",
+	testCases := []testCase{
+		{
+			name: "Event already exists",
+			event: &core.BlockchainEvent{
+				ID:         fftypes.NewUUID(),
+				Name:       "Changed",
+				Namespace:  "ns1",
```

Comment:
I see you have gone for a table test with multiple permutations of the same tests, we haven't really explored these in this codebase. At some point it would be nice to reduce code to have table tests but for now if you could keep to the same format and just have one func per test. 

I think because of the amount of mocking the table tests can be useful but sometimes difficult to manage as well